packages/react-storage/src/components/StorageBrowser/providers/configuration/useGetActionInputCallback.ts

```diff
@@ -22,21 +23,27 @@ export function useGetActionInputCallback({
   const [{ location }] = useStore();
   const { current, key } = location;
 
-  return React.useCallback(() => {
-    assertLocationData(current, ERROR_MESSAGE);
+  return React.useCallback(
+    (location?: LocationData) => {
+      const _location = current ?? location;
+      const _prefix = current ? key : location?.prefix;
+      assertLocationData(_location, ERROR_MESSAGE);
 
-    const { bucket, permission } = current;
+      const { bucket, permission, type } = _location;
 
-    return {
-      accountId,
-      bucket,
-      credentials: getCredentials({
+      return {
+        accountId,
         bucket,
-        permission,
-        prefix: key,
-      }),
-      region,
-      customEndpoint,
-    };
-  }, [accountId, current, customEndpoint, getCredentials, key, region]);
+        credentials: getCredentials({
+          bucket,
+          permission,
+          prefix: _prefix!,
```

Comment:
Not feeling great abt an explicit operator here. This is sensitive code - think we need to create/use an `assertPrefix` assertion util on line 31